### PR TITLE
avoid loading rmarkdown, packrat packages when unnecessary

### PR DESCRIPTION
**Pull request by kevinushey**
Closes https://github.com/rstudio/rstudio/issues/7265.

### CODE REVIEW

File: src/cpp/session/modules/SessionTutorial.R [hunk 1]
```diff
@@ -247,7 +247,18 @@
    
    tutorialFile <- tutorialFiles[[1]]
    contents <- readLines(tutorialFile, encoding = "UTF-8", warn = FALSE)
-   yaml <- rmarkdown:::parse_yaml_front_matter(contents)
+   
+   # find YAML header (if any). note that we avoid using rmarkdown here
+   # just because we want to avoid
```
Comment (gtritchie) [inline src/cpp/session/modules/SessionTutorial.R:252]: Incomplete comment?

File: src/cpp/session/modules/SessionTutorial.R [hunk 2]
```diff
@@ -247,7 +247,18 @@
    
    tutorialFile <- tutorialFiles[[1]]
    contents <- readLines(tutorialFile, encoding = "UTF-8", warn = FALSE)
-   yaml <- rmarkdown:::parse_yaml_front_matter(contents)
+   
+   # find YAML header (if any). note that we avoid using rmarkdown here
+   # just because we want to avoid 
+   reYamlSeparator <- "^\\s*(?:---|[.][.][.])\\s*$"
```
Comment (gtritchie) [inline src/cpp/session/modules/SessionTutorial.R:253]: Is this a common regex, i.e. we use it elsewhere for this purpose? Don't know that we have any way to share such things in our embedded R code, mainly curious if this is tried and tested vs. something new.

File: src/cpp/session/modules/SessionTutorial.R (hunk 1 above)
Comment (kevinushey) [reply src/cpp/session/modules/SessionTutorial.R]: Oops. Looks like I lost my train of thought!

File: src/cpp/session/modules/SessionTutorial.R (hunk 2 above)
Comment (kevinushey) [reply src/cpp/session/modules/SessionTutorial.R:255]: We use it for highlight rules here:

https://github.com/rstudio/rstudio/blob/3e709a0b099ee3894d4b2cd85424ae2eaf5c3e54/src/gwt/acesupport/acemode/rmarkdown_highlight_rules.js#L156-L164

And we have a similar version here:

https://github.com/rstudio/rstudio/blob/3e709a0b099ee3894d4b2cd85424ae2eaf5c3e54/src/cpp/core/YamlUtil.cpp#L32-L36

The Pandoc [spec](https://pandoc.org/MANUAL.html#extension-yaml_metadata_block) says:

> A YAML metadata block is a valid YAML object, delimited by a line of three hyphens (---) at the top and a line of three hyphens (---) or three dots (...) at the bottom.

So, strictly speaking, the second version is wrong, and so is the version in this PR -- although in practice, users almost always use `---`.

**Pull request merged**